src/factoid.rs

```diff
@@ -77,32 +77,51 @@ impl FactoidKnowledge for Brain {
 }
 
 // e.g. awoo -> awooooo or meow -> meoooow
-fn is_extension(base: &String, candidate: &String)  -> bool {
-    if base.len() == 0 && candidate.len() != 0 {
-        return false;
-    } else if base.len() == 0  && candidate.len() == 0 {
-        return true;
-    }
-
-    // blame ixi
-    let mut bc = base.chars();
+fn is_extension(base: &mut Iterator<Item=char>, mut candidate: &mut Iterator<Item=char>)  -> bool {
```

Comment:
make this `fn is_extension<T: Iterator<Item=char>, U: Iterator<Item=char>>(mut base: T, mut candidate: U) -> bool` for A Good Time (infinite length type name)